stdlib/fnmatch.pyi

```diff
@@ -1,9 +1,10 @@
+from _typeshed import GenericPath
 from collections.abc import Iterable
 from typing import AnyStr
 
 __all__ = ["filter", "fnmatch", "fnmatchcase", "translate"]
 
-def fnmatch(name: AnyStr, pat: AnyStr) -> bool: ...
-def fnmatchcase(name: AnyStr, pat: AnyStr) -> bool: ...
-def filter(names: Iterable[AnyStr], pat: AnyStr) -> list[AnyStr]: ...
+def fnmatch(name: GenericPath[AnyStr], pat: AnyStr) -> bool: ...
+def fnmatchcase(name: GenericPath[AnyStr], pat: AnyStr) -> bool: ...
```

Comment:
This doesn't work

```
>>> import fnmatch
>>> from pathlib import Path
>>> fnmatch.fnmatchcase(Path("hi"), "x")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jelle/Library/Application Support/uv/python/cpython-3.8.19-macos-aarch64-none/lib/python3.8/fnmatch.py", line 71, in fnmatchcase
    return match(name) is not None
TypeError: expected string or bytes-like object
```

The other functions do currently work with paths, but I'm not sure we should change the stubs; it feels accidental, and the module documentation talks about "path strings".

Reply:
pathlib's trailing slash normalisation can also be undesirable in fnmatch like situations